zone/spells.cpp

```diff
@@ -6770,3 +6771,11 @@ void Mob::SetHP(int64 hp)
 
 	current_hp = hp;
 }
+
+bool Mob::CheckWaterLoS(Mob* los_attacker, Mob* los_target) // checks if both attacker and target are both in or out of the water
+{
+	if (!RuleB(Spells, WaterMatchRequiredForLoS)) { // if rule is set to false, bypass check
+		return true;
+	}
+	return zone->watermap->InLiquid(los_attacker->GetPosition()) == zone->watermap->InLiquid(los_target->GetPosition());
```

Comment:
Same here re: checking `zone->watermap` isn't nullptr, `spell_target` is checked by caller and I think `CheckLosFN` would have crashed before this if something deeper was going on with a trashed Mob pointer

Reply:
This one should also take a single argument.